apps/wpcom-block-editor/src/common/disable-nux-tour.js

```diff
@@ -0,0 +1,6 @@
+/**
+ * External dependencies
+ */
+import { dispatch } from '@wordpress/data';
+
+dispatch( 'core/nux' ).disableTips();
```

Comment:
We'll likely run into timing issues here if we call this on script load.

Reply:
@youknowriad  @aduth @ellatrix 

We're trying to disable tips on editor load. Is the above okay, or do we need to guard against the store loaded check with something like:

```js
import { select, subscribe, dispatch } from '@wordpress/data';

const unsubscribe = subscribe( () => {
	const nuxInitialized = select( 'core/nux' ) !== undefined;
	if ( ! nuxInitialized ) {
		return;
	}
	unsubscribe();
	dispatch( 'core/nux' ).disableTips();
} );
```

Reply:
If possible, you should define a dependency on the `@wordpress/nux` module. Doing so should ensure the store is registered.

This is how it's done in the editor:

https://github.com/WordPress/gutenberg/blob/f1bbc004734ea157dd72ed5bf1f808c10efc0e84/packages/editor/src/index.js#L8

The NUX store is registered as a side effect of the module load:

https://github.com/WordPress/gutenberg/blob/f1bbc004734ea157dd72ed5bf1f808c10efc0e84/packages/nux/src/index.js#L4
https://github.com/WordPress/gutenberg/blob/f1bbc004734ea157dd72ed5bf1f808c10efc0e84/packages/nux/src/store/index.js#L13

Reply:
Ah so it'd look like:

```jsx
import { dispatch } from '@wordpress/data';
import '@wordpress/nux'; //ensure nux store loads

dispatch( 'core/nux' ).disableTips();
```

Did we previously run into trouble with store loading/versioning @mmtr @jsnajdr ?

Reply:
I guess to add to that @aduth assuming we don't accidentally pull in multiple versions of the nux package, is it safe to import the nux package more than once in different files?

Reply:
Good point @gwwar. I think `registerStore` should check the name is not used by other store. Maybe it's already doing that, but I'd try to avoid multiple imports just in case. If we ever need this import in another file, we should move it to a file that is higher in the hierarchy (like `common/index.js`)

Reply:
Anyway, since this package is intended to be loaded in a WordPress environment, we generate the builds with `@automattic/calypso-build` and the [`--env.WP` option](https://github.com/Automattic/wp-calypso/blob/12e94e5f7e30a7871fdce0f2c9eff689e339c440/apps/wpcom-block-editor/package.json#L19) that externalizes all the `@wordpress/<package-name>` dependencies.

So when we create the WP.com patch that updates the build, we need to make sure to add the `wp-nux` dependency when enqueuing the script (see `gutenberg-wpcom/index.php#L69`).

Reply:
> I guess to add to that @aduth assuming we don't accidentally pull in multiple versions of the nux package, is it safe to import the nux package more than once in different files?

I think it should be fine. Ultimately it depends on the guarantees offered by the build tool. Unsure if factors like [scope hoisting](https://medium.com/webpack/brief-introduction-to-scope-hoisting-in-webpack-8435084c171f) come in to play. I expect Webpack tries to ensure the code is only run once.

While it doesn't help your immediate need, there's some early efforts to expose the underlying store configurations for some (but not all, including NUX) stores ([example](https://github.com/WordPress/gutenberg/blob/514e61298d5c056ef6ec4d3859b959dc46735dec/packages/block-editor/src/store/index.js#L20-L25)). I don't think there'd be an objection to just applying this pattern for all stores, and sooner than later.